packages/security_center/lib/disk_encryption/disk_encryption_page.dart

```diff
@@ -33,109 +33,165 @@ class _Body extends ConsumerWidget {
+          Text(
+            l10n.diskEncryptionPageRecoveryKey,
+            style: Theme.of(context).textTheme.titleMedium,
+            textAlign: TextAlign.left,
+          ),
+          const SizedBox(height: 16),
+          MarkdownText(
+            '${l10n.diskEncryptionPageStoreYourKey} ${l10n.diskEncryptionPageLearnMore.link(_learnMoreUrl)}',
+          ),
+          const SizedBox(height: 16),
+          Wrap(
+            spacing: 16,
+            children: [
+              OutlinedButton(
+                onPressed: () {
+                  showCheckRecoveryKeyDialog(context);
+                },
+                child: Text(l10n.diskEncryptionPageCheckKey),
+              ),
+              OutlinedButton(
+                onPressed: () {
+                  showReplaceRecoveryKeyDialog(context);
+                },
+                child: Text(l10n.diskEncryptionPageReplaceButton),
+              ),
+            ],
+          ),
+          const SizedBox(height: 16),
+          // TPM Authentication specific content
+          switch (data) {
+            AuthMode.pin => Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    l10n.recoveryKeyPinHeader,
+                    style: Theme.of(context).textTheme.titleMedium,
+                  ),
+                  const SizedBox(height: 8),
+                  Text(l10n.recoveryKeyPinBody),
+                  const SizedBox(height: 16),
+                  OutlinedButton(
+                    onPressed: () {
+                      ref
+                          .read(changeAuthDialogModelProvider.notifier)
+                          .authMode = AuthMode.pin;
+                      showChangeAuthDialog(context, AuthMode.pin);
+                    },
+                    child: Text(l10n.recoveryKeyPinButton),
+                  ),
+                ],
+              ),
+            AuthMode.passphrase => Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    l10n.recoveryKeyPassphraseHeader,
+                    style: Theme.of(context).textTheme.titleMedium,
+                  ),
+                  const SizedBox(height: 8),
+                  Text(l10n.recoveryKeyPassphraseBody),
+                  const SizedBox(height: 16),
+                  OutlinedButton(
+                    onPressed: () {
+                      ref
+                          .read(changeAuthDialogModelProvider.notifier)
+                          .authMode = AuthMode.passphrase;
+                      showChangeAuthDialog(context, AuthMode.passphrase);
+                    },
+                    child: Text(l10n.recoveryKeyPassphraseButton),
+                  ),
+                ],
+              ),
+            _ => const SizedBox.shrink(),
+          },
+        ],
       ),
+      error: (e, stack) {
+        try {
```

Comment:
Can this just be a `switch` expression? I don't think throwing and immediately catching the exception is necessary here